### ISSUE TITLE
stubtest flags PEP 661 `sentinel` defaults, but not `object()` markers

### ISSUE THREAD

**Issue by ariebovenberg** [python/mypy#22013]
**Bug Report**

I was eager to try PEP661 support on `master`, but found what might be an oversight in `stubtest`. It flags sentinel usage against the stub's parameter type, while it does allow `object`-style marker objects. Since PEP661's sentinel is now the canonical way to write such a marker, I'd expect it to be allowed in such places:

**To Reproduce**

```python
# foo.py
from typing_extensions import sentinel
_MISSING = sentinel("_MISSING")


def f(x=_MISSING):
    pass
```

```python
# foo.pyi
def f(x: int = ...) -> None: ...
```

```
python -m mypy.stubtest foo
```

**Expected Behavior**

No errors, the same as with `_MISSING = object()`:

```
Success: no issues found in 1 module
```

**Actual Behavior**

```
error: foo.f is inconsistent, runtime parameter "x" has a default value of type typing_extensions.sentinel, which is incompatible with stub parameter type int
Stub: in file foo.pyi:1
def (x: int =)
Runtime: in file foo.py:5
def (x=_MISSING)

Found 1 error (checked 1 module)
```

I get the same on Python 3.15 beta.

As described above, adding the sentinel to the stub's type (`x: int | _MISSING = ...`) isn't an option.

The exemption seems to be in `_verify_arg_default_value`:

```python
# Avoid false positives for marker objects
and type(runtime_arg.default) is not object
```

It could be adjusted to include `sentinel` the same way.

**My Environment**

- Mypy version used: master at 8d30064580c4273542f56ab2e9ed6a17caeea8ac (2.4.0+dev)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.14.7 (`typing_extensions` 4.16.0)

**Comment by JelleZijlstra** [comment]
I think we didn't consider stubtest when adding support (cc @edgarrmondragon). PR welcome!

**Comment by edgarrmondragon** [comment]
I'm taking a look

UPDATE: https://github.com/python/mypy/issues/22017, but I'm curious if we're missing anything related to stubs for sentinels?

**Comment by JelleZijlstra** [comment]
Your PR looks right, maybe we also need to change something in stubgen though?

**Comment by edgarrmondragon** [comment]
> Your PR looks right, maybe we also need to change something in stubgen though?

Thanks for the hint, see https://github.com/python/mypy/pull/22019.
